### ISSUE TITLE
reference: silent crash

### ISSUE THREAD

**Issue by Mouvedia** [axr/core#120]
xml

```
<state>
    <parent>
        <child1>test1</child1>
    </parent>
</state>
```

this crashes:

```
state
{
    parent
    {
        background: #0;
        child1 { background: ref(background of @root parent); }
    }
}
```

this doesn't:

```
state
{
    parent
    {
        background: ref(background of @root parent child1);
        child1 { background: #0; }
    }
}
```

this does:

```
state
{
    background: #0;
    parent
    {
        background: ref(background of @root);
    }
}
```

and this too:

```
state
{
    background: ref(background of parent child1);
    parent
    {
        child1 { background: #0; }
    }
}
```

OS: Windows XP

**Comment by veosotano** [comment]
This got fixed by 918ec322dfc3af42c2bed3ccc435bcd06f5ab8ab

**Comment by Mouvedia** [comment]
So you are actually using `@this` on the root element when you use `@root`?
